concourse/pipeline.yml

```diff
@@ -5,117 +5,133 @@ resource_types:
+        - concourse/Dockerfile
         - Gemfile*
-        - .ruby-version
-        - spec/
 
   - name: every-day
     type: time
     source:
       interval: 24h
 
-  - name: govuk-coronavirus-vulnerable-people-form-travis-build
-    type: travis
-    icon: sync
-    source:
-      repository: alphagov/govuk-coronavirus-vulnerable-people-form
-      travis-token: ((travis-api-token))
-      branch: master
-      pro: true
-
-  - name: feature-tests-image
+  - name: tests-image
     type: docker-image
     icon: docker
     source:
       repository: ((readonly_private_ecr_repo_url))
-      tag: govuk-coronavirus-vulnerable-people-form-feature-tests
+      tag: govuk-coronavirus-vulnerable-people-tests-image
 
 jobs:
   - name: update-pipeline
     plan:
-      - get: govuk-coronavirus-vulnerable-people-form
+      - get: git-master
         trigger: true
       - set_pipeline: govuk-corona-vulnerable-people-form
-        file: govuk-coronavirus-vulnerable-people-form/concourse/pipeline.yml
+        file: git-master/concourse/pipeline.yml
 
-  - name: build-feature-tests-image
+  - name: build-tests-image
     serial: true
+    build_logs_to_retain: 100
     plan:
       - get: every-day
         trigger: true
-      - get: govuk-coronavirus-vulnerable-people-form-feature-tests
+      - get: git-master-gems-dockerfile
         trigger: true
-      - put: feature-tests-image
+      - put: tests-image
         params:
-          build: govuk-coronavirus-vulnerable-people-form-feature-tests
-          dockerfile: govuk-coronavirus-vulnerable-people-form-feature-tests/spec/Dockerfile
+          build: git-master-gems-dockerfile
+          dockerfile: git-master-gems-dockerfile/concourse/Dockerfile
+
+  - name: run-quality-checks
+    serial: true
+    plan:
+      - get: tests-image
+        passed: [build-tests-image]
+        trigger: true
+      - get: git-master
+        trigger: true
+      - task: run-tests-task
+        image: tests-image
+        config:
+          inputs:
+            - name: git-master
+          platform: linux
+          run:
+            dir: git-master
+            path: bash
+            args:
+              - -c
+              - |
+                set -eu
+                wget https://s3.eu-central-1.amazonaws.com/dynamodb-local-frankfurt/dynamodb_local_latest.tar.gz
+                sha256sum -c <<< "d77f5371c21868dd489b53782619b7f2e418d4b1068947a28d80082052ca1d64 dynamodb_local_latest.tar.gz"
+                tar -xzf dynamodb_local_latest.tar.gz
+                java -jar DynamoDBLocal.jar>&1 &
+                export RAILS_ENV=test
+                bundle install
+                bundle exec rake
 
   - name: deploy-to-staging
     serial: true
     plan:
-      - get: govuk-coronavirus-vulnerable-people-form-travis-build
+      - get: git-master
         trigger: true
-      - get: govuk-coronavirus-vulnerable-people-form
+        passed: [run-quality-checks]
       - task: deploy-to-paas
         config:
```

Comment:
We indented this properly, but then the concourse validator started complaining file: was invalid.

We've got this on a few other pipelines. I'm not sure what the syntax or indentation should be. seems really wrong to have a `config:` on its own line and everything else at the same level too.

Might need to also check this on:
- [find support](https://github.com/alphagov/govuk-coronavirus-find-support/blob/master/concourse/pipeline.yml#L93-L105)
- [biz form](https://github.com/alphagov/govuk-coronavirus-business-volunteer-form/blob/master/concourse/pipeline.yml#L75-L96)

comparing them i think it's a redundant key that's just rendering to `nil`. It was only there because we had some copy paste stuff from some verify concourse pipelines. I think it can be removed... and should be elsewhere too!

Reply:
What's "the Concourse validator"?

Reply:
When you run `set-pipeline` in fly it takes in the provided YAML and runs some validation checks. I'm pretty sure it's the same thing as if you run `validate-pipeline` [see here](https://concourse-ci.org/setting-pipelines.html#fly-validate-pipeline). It'll tell you if you have any obvious syntax errors in your `pipeline.yml`.

I've also had an odd case where it made it past the validatior, but then i guess something in concourse compiles it into JSON. Got a really unhelpful "JSON is invalid" kind-of error without a line number. Turned out I'd indented something wrong in the yml. So i guess that's another kind of validation...

Reply:
Ohhh, right, yes!